File: Sprint02-Backend/src/main/java/com/br/saks/Sprint02/sprint02/model/Usuario.java
package com.br.saks.Sprint02.sprint02.model;

import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

@Data
@Entity
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 200)
    private String email;

    @Column(nullable = false, length = 200)
    private String senha;

    @Column(nullable = false, length = 200)
    private String nome;

    @Column(length = 1)
    private Integer status;
    
    @JsonIgnore
    @OneToMany(mappedBy = "usuario")
    private List<Objetivo> objetivos;
    
}

File: Sprint02-Frontend/lib/entities/usuario.dart
import 'Objetivo.dart';

class Usuario {
  int? id;
  String? nome;
  String? email;
  String? senha;
  String? confirmarSenha;
  List<dynamic>? objetivos;


  Usuario(
      {this.id, this.nome, this.email, this.senha, this.confirmarSenha, this.objetivos});

  factory Usuario.fromJson(Map<String, dynamic> json){
    Usuario usuario = Usuario(
      id: json['id'],
      nome: json['nome'],
      email: json['email'],
      senha: json['senha'],
      confirmarSenha: '',
    );
    if (json['objetivos'] != null) {
      usuario.objetivos = json['objetivos']
          .map((objetivo) => Objetivo.fromJson(objetivo))
          .toList();
    }
    return usuario;
  }
}